lib/max_subarray.rb

```diff
@@ -1,8 +1,25 @@
 
-# Time Complexity: ?
-# Space Complexity: ?
+# Time Complexity: O(n)
+# Space Complexity: O(1)
 def max_sub_array(nums)
-    return 0 if nums == nil
+    return 0 if nums == nil || nums.empty?
     
-    raise NotImplementedError, "Method not implemented yet!"
+    max = nums[0]
+    current = nums[0]
+    
+    nums.each do |index|
```

Comment:
index here won't be an index into the array, it will be a value.  

Maybe you meant:
```suggestion
    nums.length.time do |index|
```